Profiler_Cpp_Test/tests/ConfigTest.cpp

```diff
@@ -0,0 +1,169 @@
+#include <sstream>
+#include "CppUnitTest.h"
+#include "config/Config.h"
+
+using namespace Microsoft::VisualStudio::CppUnitTestFramework;
+
+TEST_CLASS(ConfigTest)
+{
+public:
+
+	TEST_METHOD(EmptyConfigAndNoEnvironment)
+	{
+		Config config = parse(R"()", [](std::string suffix) -> std::string { return "";});
+
+		Assert::AreEqual(size_t(0), config.getProblems().size(), L"number of problems");
+		Assert::AreEqual(true, config.isEnabled(), L"default value should be enabled");
+		Assert::AreEqual(false, config.shouldIgnoreExceptions(), L"default value should be to not ignore exceptions");
+		Assert::AreEqual(size_t(0), config.getEagerness(), L"default value should be no eagerness");
+	}
+
+	TEST_METHOD(EnvironmentMustBeRespected)
+	{
+		Config config = parse(R"()", [](std::string suffix) -> std::string {
+			if (StringUtils::uppercase(suffix) == "IGNORE_EXCEPTIONS") {
+				return "1";
+			}
+			return "";
+		});
+
+		Assert::AreEqual(true, config.shouldIgnoreExceptions(), L"should use value from environment");
+	}
+
+	TEST_METHOD(ConfigFileMustBeRespected)
+	{
+		Config config = parse(R"(
+match:
+  - process: ".*"
+    profiler:
+      ignore_exceptions: true
+)", [](std::string suffix) -> std::string { return ""; });
+
+		Assert::AreEqual(true, config.shouldIgnoreExceptions(), L"should use value from config");
+	}
+
+	TEST_METHOD(OnlyMatchingSectionsAreApplied)
+	{
+		Config config = parse(R"(
+match:
+  - process: ".*doesnt-match"
+    profiler:
+      ignore_exceptions: true
+)", [](std::string suffix) -> std::string { return ""; });
+
+		Assert::AreEqual(false, config.shouldIgnoreExceptions(), L"should be the default value");
+	}
+
+	TEST_METHOD(NoProcessFieldMeansMatchAnyProcess)
+	{
+		Config config = parse(R"(
+match:
+  - profiler:
+      ignore_exceptions: true
+)", [](std::string suffix) -> std::string { return ""; });
+
+		Assert::AreEqual(true, config.shouldIgnoreExceptions(), L"should be the config value");
+	}
+
+	TEST_METHOD(EnvironmentTrumpsConfig)
```

Comment:
Hm, I wouldn't have said so since they test different aspects:

- is the environment considered at all?
- is the environment more important than the config file?

I the first succeeds but the second fails I already know more than if I can only see that the second fails IMO

Reply:
Fair enough.